### DISCUSSION TITLE
migu的音源失效了吗

### DISCUSSION THREAD

**Discussion by wanlala** [1715173329/UnblockNeteaseMusic#707] [Q&A] [answered]
环境：ubuntu22.04
启动命令：/usr/bin/node /opt/server-enhanced/app.js -p 5221:5222 -o migu pyncmd
打扰一下！
蛮奇怪的 昨天我还能用 然后手滑删掉了整个opt文件夹
源码是部署在opt目录下的
然后重新安装nodejs npm之后再部署
migu的源就不能用了ww  （别的还能用
而且咪咕有的歌可以解析 有的之前解析成功过的歌却不行了
`(node:14866) UnhandledPromiseRejectionWarning: TypeError: AggregateError is not a constructor
    at /opt/server-enhanced/precompiled/app.js:13405:33
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:14866) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
(node:14866) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    at emitDeprecationWarning (internal/process/promises.js:161:11)
    at processPromiseRejections (internal/process/promises.js:213:13)
    at processTicksAndRejections (internal/process/task_queues.js:98:32)
`

**Comment by pan93412** [comment]
cc @1715173329 

如果 Node.js 已經是 v16 以上的話，就直接 `node src/app.js` 吧

**Comment by 1715173329** [comment, marked as answer]
此音源正常工作中，请参见 readme 更改日志等级为 debug 以查看详细信息。

**Reply by wanlala** [reply]
`DEBUG: (server) TUNNEL
    url: "localhost:5222"
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/find"
DEBUG: (cache) 418602085 did not hit. Storing the execution result...
    logKey: 418602085
    cacheStorageId: "provider/find"
DEBUG: (request) Start requesting https://music.163.com/api/song/detail?ids=[418602085]
DEBUG: (provider/match) Getting the audio...
    source: "migu"
    info: {
      "name": "英雄",
      "id": 418602085,
      "alias": [
        "游戏《英雄联盟》中国品牌主题曲"
      ],
      "duration": 201746,
      "album": {
        "name": "周杰伦的床边故事",
        "id": 34720827
      },
      "artists": [
        {
          "name": "周杰伦",
          "id": 6452
        }
      ],
      "keyword": "英雄 - 周杰伦"
    }
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/migu"
DEBUG: (cache) Something did not hit. Storing the execution result...
    logKey: "Something"
    cacheStorageId: "provider/migu"
DEBUG: (request) Start requesting https://m.music.migu.cn/migu/remoting/scr_search_tag?keyword=%E8%8B%B1%E9%9B%84%20-%20%E5%91%A8%E6%9D%B0%E4%BC%A6&type=2&rows=20&pgc=1
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=28669236657441255&netType=01&resourceType=E&songId=1004202154&toneFlag=HQ
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=1130263294028091&netType=01&resourceType=E&songId=1004202154&toneFlag=PQ
DEBUG: (server) TUNNEL
    url: "localhost:5222"
(node:38333) UnhandledPromiseRejectionWarning: TypeError: AggregateError is not a constructor
    at /opt/server-enhanced/precompiled/app.js:13405:33
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:38333) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:38333) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/find"
DEBUG: (cache) 418602085 hit!
    logKey: 418602085
    cacheStorageId: "provider/find"
DEBUG: (provider/match) Getting the audio...
    source: "migu"
    info: {
      "name": "英雄",
      "id": 418602085,
      "alias": [
        "游戏《英雄联盟》中国品牌主题曲"
      ],
      "duration": 201746,
      "album": {
        "name": "周杰伦的床边故事",
        "id": 34720827
      },
      "artists": [
        {
          "name": "周杰伦",
          "id": 6452
        }
      ],
      "keyword": "英雄 - 周杰伦"
    }
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/migu"
DEBUG: (cache) Something hit!
    logKey: "Something"
    cacheStorageId: "provider/migu"
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=15801054144807014&netType=01&resourceType=E&songId=1004202154&toneFlag=HQ
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=26862411962200006&netType=01&resourceType=E&songId=1004202154&toneFlag=PQ
(node:38333) UnhandledPromiseRejectionWarning: TypeError: AggregateError is not a constructor
    at /opt/server-enhanced/precompiled/app.js:13405:33
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:38333) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 6)
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/find"
DEBUG: (cache) 418602085 hit!
    logKey: 418602085
    cacheStorageId: "provider/find"
DEBUG: (provider/match) Getting the audio...
    source: "migu"
    info: {
      "name": "英雄",
      "id": 418602085,
      "alias": [
        "游戏《英雄联盟》中国品牌主题曲"
      ],
      "duration": 201746,
      "album": {
        "name": "周杰伦的床边故事",
        "id": 34720827
      },
      "artists": [
        {
          "name": "周杰伦",
          "id": 6452
        }
      ],
      "keyword": "英雄 - 周杰伦"
    }
DEBUG: (cache) Cleaning up the expired caches...
    cacheStorageId: "provider/migu"
DEBUG: (cache) Something hit!
    logKey: "Something"
    cacheStorageId: "provider/migu"
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=421783927503969&netType=01&resourceType=E&songId=1004202154&toneFlag=HQ
DEBUG: (request) Start requesting https://app.c.nf.migu.cn/MIGUM2.0/strategy/listen-url/v2.2?lowerQualityContentId=9830727375014359&netType=01&resourceType=E&songId=1004202154&toneFlag=PQ
(node:38333) UnhandledPromiseRejectionWarning: TypeError: AggregateError is not a constructor
    at /opt/server-enhanced/precompiled/app.js:13405:33
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:38333) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 9)
`
报错如上
